mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java

```diff
@@ -68,12 +68,12 @@ public static IndexableMessage from(MailboxMessage message, List<User> users, Te
     }
 
     private void setFlattenedAttachments(MimePart parsingResult, IndexAttachments indexAttachments) {
-        if (indexAttachments.equals(IndexAttachments.YES)) {
-            attachments = parsingResult.getAttachmentsStream()
-                    .collect(Collectors.toList());
-        } else {
-            attachments = ImmutableList.of();
-        }
+        List<MimePart> mimeparts = parsingResult.getAttachmentsStream()
+                .collect(Collectors.toList());
+
+        hasAttachment = mimeparts.stream().anyMatch(IndexableMessage::isAttachmentMimepart);
+
+        this.attachments = IndexAttachments.YES.equals(indexAttachments) ? mimeparts : ImmutableList.of();
```

Comment:
they could be readable but the real argument IMO is that a ternary operator is better in its own function.